File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 8/Com Pacotes/Pessoa.js
module.exports = class Pessoa {
    constructor(nome, endereco, idade, listaDisc){
        this.nome = nome
        this.endereco = endereco
        this.idade = idade
        this.listaDisc = listaDisc
    }

    insereDisc(disc){
        this.listaDisc.push(disc)
    }
}

File: 04-ORIENTACAO_A_OBJETOS/Java/Exercicio 8/Pessoa.java

import java.util.ArrayList;

public abstract class Pessoa {
    private String nome;
    private String endereco;
    private int idade;
    private ArrayList<Disciplina> listaDisc = new ArrayList();
    
    Pessoa(String nome, String endereco, int idade, ArrayList<Disciplina> listaDisc){
        this.nome = nome;
        this.endereco = endereco;
        this.idade = idade;
        this.listaDisc = listaDisc;
    }
    
    String getNome(){
        return this.nome;
    }
    
    String getEndereco(){
        return this.endereco;
    }
    
    int getIdade(){
        return this.idade;
    }
    
    ArrayList<Disciplina> getListaDisc(){
        return this.listaDisc;
    }
    
    void insereDisc(Disciplina disc){
        this.listaDisc.add(disc);
    }
    
    abstract void printDescricao();

}
